serve/tests/unittest/test_sampler.py

```diff
@@ -31,132 +34,194 @@ def get_sampling_state(sampling_params, past_output_tokens=None, prompt_masks=No
     return sampling_state
 
 
-def test_temperature(temp=0, batch_size=1):
-    shape = (batch_size, vocab_size)
-    logits = torch.rand(shape, dtype=dtype, device=dev)
-    sampling_param = SamplingParams(temperature=temp)
-
-    sampling_state = get_sampling_state([sampling_param])
-
-    expected = logits / temp if abs(temp) > SAMPLING_EPS else logits
-    new_logits = adjust_logits(logits, sampling_state, vocab_size)
-    assert torch.allclose(expected, new_logits)
+def test_temperature_checker():
+    # temperature must be in [0, 2]
+    get_sampling_state([SamplingParams(temperature=0.0)])
+    get_sampling_state([SamplingParams(temperature=0.8)])
+    get_sampling_state([SamplingParams(temperature=1.3)])
+    get_sampling_state([SamplingParams(temperature=2.0)])
 
-
-def test_logit_bias_checker():
-    # logit bias must be [-100, 100]
     with pytest.raises(ValueError):
-        logit_bias = {1: 2, 3: 105, 2: 2}
-        sampling_param = SamplingParams(logit_bias=logit_bias)
-        get_sampling_state([sampling_param])
+        get_sampling_state([SamplingParams(temperature=-0.1)])
 
     with pytest.raises(ValueError):
-        logit_bias = {1: 99, 3: -101, 2: 2}
-        sampling_param = SamplingParams(logit_bias=logit_bias)
-        get_sampling_state([sampling_param])
+        get_sampling_state([SamplingParams(temperature=2.1)])
+
+@pytest.mark.parametrize("batch_size", [1, 4, 8, 12])
+def test_temperature(batch_size: int):
+    shape = (batch_size, vocab_size)
+    logits = torch.rand(shape, dtype=dtype, device=dev)
+    temperature = [0, 0.5, 1.0, 1.5, 2.0]
+    for batch_temp in permutations(temperature, batch_size):
+        sampling_params = [
+            SamplingParams(temperature=val)
+            for val in batch_temp
+        ]
+        expected = []
+        for idx, val in enumerate(batch_temp):
+            expected.append(logits[idx] / val if abs(val) > SAMPLING_EPS else logits[idx])
+        sampling_state = get_sampling_state(sampling_params)
+        new_logits = adjust_logits(logits, sampling_state, vocab_size)
+        for idx, response in enumerate(new_logits):
+            assert torch.allclose(expected[idx], response)
 
-    logit_bias = {1: 100, 3: -100, 2: 2}
-    sampling_param = SamplingParams(logit_bias=logit_bias)
-    get_sampling_state([sampling_param])
 
+def test_logit_bias_checker():
+    # logit bias values must be [-100, 100]
+    get_sampling_state([SamplingParams(logit_bias={1: 100, 3: -100, 2: 2})])
+    get_sampling_state([SamplingParams(logit_bias={34: 0, 23: -0.5})])
     # TODO(@team): it seems like the valid range is [1,vocab_size]. Double check.
```

Comment:
I didn't find any descriptive API reference for indices in logit_bias, but figured that in transformers similar parameter sequence_bias [must be](https://github.com/huggingface/transformers/blob/fc37f38915372c15992b540dfcbbe00a916d4fc6/src/transformers/generation/logits_process.py#L1078) >= 0, but if not then error message says it has to be [positive](https://github.com/huggingface/transformers/blob/fc37f38915372c15992b540dfcbbe00a916d4fc6/src/transformers/generation/logits_process.py#L1083) so the usage of 0 is not quite clear here. I guess it can be also used since these are all indices and there is a token (`<unk>`) with id 0 so I changed range to `[0, vocab_size)` in the last commits

Reply:
I see. Can we follow-up about this after this PR? I think this can be a rough edge so would like to address this before we run into the issue with customers. Maybe we can do deep-dive `transformers` or `TGI` to match our behavior.